File: 01.Basics/client-go/list_pods.go

package main

import (
    "context"
    "fmt"
    "k8s.io/client-go/kubernetes"
    "k8s.io/client-go/tools/clientcmd"
    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func main() {
    config, err := clientcmd.BuildConfigFromFlags("", clientcmd.RecommendedHomeFile)
    if err != nil {
        panic(err.Error())
    }

    clientset, err := kubernetes.NewForConfig(config)
    if err != nil {
        panic(err.Error())
    }

    pods, err := clientset.CoreV1().Pods("default").List(context.TODO(), metav1.ListOptions{})
    if err != nil {
        panic(err.Error())
    }

    fmt.Println("Listing pods in the default namespace:")
    for _, pod := range pods.Items {
        fmt.Printf("%s\t%s\t%s\n", pod.Name, pod.Status.Phase, pod.Spec.Containers[0].Image)
    }
}

File: 01.Basics/python/list_pods.py
from kubernetes import client, config

# Load kube config from default location
config.load_kube_config()

v1 = client.CoreV1Api()
print("Listing pods in the default namespace:")
for pod in v1.list_namespaced_pod(namespace="default").items:
    print(f"{pod.metadata.name}\t{pod.status.phase}\t{pod.spec.containers[0].image}")
